core/src/main/java/hudson/PluginManager.java

```diff
@@ -1955,6 +1955,8 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
                 }
             } catch (MalformedURLException e) {
                 return FormValidation.error(e.getMessage());
+            } catch (URISyntaxException e) {
+                throw new URISyntaxException(e.getReason(), e.getInput());
```

Comment:
you need to handle this as appropriate for the function - do not rethrow this.

in other words the handling of this exception is returning a `FormValidation` error with the message so that the user can understand they have entered giberish into the form

Reply:
Replaced with below as explained above
```
} catch (MalformedURLException |URISyntaxException e) {
                return FormValidation.error(e.getMessage());
            }
```